processing/src/main/java/org/apache/druid/query/dimension/ListFilteredDimensionSpec.java

```diff
@@ -92,15 +92,17 @@ public static IdMapping buildAllowListIdMapping(
       IndexedGetter<String> fn
   )
   {
-    final IdMapping.Builder builder = IdMapping.Builder.ofCardinality(values.size());
+    IdMapping.Builder builder;
     if (idLookup != null) {
+      builder = IdMapping.Builder.ofCardinality(values.size());
```

Comment:
i think this could also be impacted the same way as the other branch, so I don't really think this fixes things (though it does the virtual column since it uses the 'else' branch here

Reply:
I was thinking this might get into similar troubles - but then noticed that: this branch uses a foreach on the `values` ; so it may make at-most `|values|` calls to `addMapping` - so I think its safe to leave it like this.

...hmm...but I'm wondering if I've misunderstood your comment: what did you mean by "impacted the same way" - the exception issue; or some perf stuff?